python/ray/tests/kuberay/test_autoscaling_e2e.py

```diff
@@ -165,46 +186,47 @@ def testAutoscaling(self):
         - One Ray head pod
             - Autoscaler: .25 CPU, .5 Gi memory
             - Ray node: .5 CPU, .5 Gi memeory
-        - Two Worker pods
+        - Three Worker pods
             - Ray node: .5 CPU, .5 Gi memory
-        Total: 1.75 CPU, 2 Gi memory.
+        Total: 2.25 CPU, 2.5 Gi memory.
 
         Including operator and system pods, the total CPU requested is around 3.
 
         The cpu LIMIT of each Ray container is 1.
         The `num-cpus` arg to Ray start is 1 for each Ray container; thus Ray accounts
         1 CPU for each Ray node in the test.
         """
-        # Cluster-creation
+        # Cluster creation
         logger.info("Creating a RayCluster with no worker pods.")
         self._apply_ray_cr(min_replicas=0, replicas=0)
 
         logger.info("Confirming presence of head.")
         wait_for_pods(goal_num_pods=1, namespace="default")
-        head_pod = get_pod(
-            pod_name_filter="raycluster-complete-head", namespace="default"
-        )
 
         logger.info("Waiting for head pod to start Running.")
-        wait_for_pod_to_start(head_pod, namespace="default")
+        wait_for_pod_to_start(
+            pod_name_filter="raycluster-complete-head", namespace="default"
+        )
         logger.info("Confirming Ray is up on the head pod.")
-        wait_for_ray_health(head_pod, namespace="default")
+        wait_for_ray_health(
+            pod_name_filter="raycluster-complete-head", namespace="default"
+        )
 
+        head_pod = get_pod(
+            pod_name_filter="raycluster-complete-head", namespace="default"
+        )
         # Scale-up
         logger.info("Scaling up to one worker via Ray resource request.")
-        scale_script = (
-            "import ray;"
-            'ray.init("auto");'
-            "ray.autoscaler.sdk.request_resources(num_cpus=2)"
-        )
         # The request for 2 cpus should give us a 1-cpu head (already present) and a
         # 1-cpu worker (will await scale-up).
-        kubectl_exec(
-            command=["python", "-c", scale_script],
+        kubectl_exec_python_script(
+            script_name="scale_up.py",
             pod=head_pod,
             container="ray-head",
             namespace="default",
         )
+        # TODO (Dmitri) Use Ray Client and/or Ray Job submission API to submit
+        # instead of `kubectl exec`.
```

Comment:
In a follow up PR, we can actually use all three methods in this test: Ray client, job submission, kubectl exec.